File: 0009-palindrome-number/0009-palindrome-number.go
/*
func isPalindrome(x int) bool {
    xStr := strconv.Itoa(x)
    return palindrome(xStr)
}

func palindrome(s string) bool {
    if len(s) <= 1 {
        return true
    }
    if s[0] != s[len(s)-1] {
        return false
    }
    return palindrome(s[1:len(s)-1])
}
*/

func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }
    
    before := x
    after := 0
    for 0 < x {
        after *= 10
        after += x % 10
        x /= 10
    }
    
    return before == after
}

File: 0009-palindrome-number/0009-palindrome-number.py
"""
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x_str = str(x)
        return self.check_palindrome(x_str)
    
    def check_palindrome(self, s: str) -> bool:
        if len(s) <= 1:
            return True
        if s[0] != s[-1]:
            return False
        return self.check_palindrome(s[1:-1])
            
"""

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        before = x
        after = 0
        while x > 0:
            after *= 10
            after += (x % 10)
            x //= 10
        return before == after    